### ISSUE TITLE
reportPossiblyUnboundVariable not defaulting to "error"

### ISSUE THREAD

**Issue by lexicalunit** [microsoft/pyright#9299]
**Describe the bug**
At runtime we encountered a crash in our application today because of an unbound variable in our code. We use `pyright` in our CI/CD, so we were surprised that an unbound error wasn't caught by it before the code could be merged into production. Looking at the documentation for `pyright` we found the option `reportPossiblyUnboundVariable`. We then set that to `"error"` in our `pyproject.toml` and re-ran `pyright` on our codebase. It then found the issue in the code. This seems weird because this option, `reportPossiblyUnboundVariable`, is supposed to be `"error"` by default according to the documentation: https://github.com/microsoft/pyright/blob/main/docs/configuration.md

> reportPossiblyUnboundVariable [boolean or string, optional]: Generate or suppress diagnostics for variables that are possibly unbound on some code paths. The default value for this setting is "error".

**Code or Screenshots**
For example:
```python
import random

def whatever():
    return random.randint(0, 1)
    
if whatever():
    foo = 5

print(foo)  # possibly unbound
```

**VS Code extension or command-line**
We are using the command line. Our `pyright` version is `1.1.380`.

Here is our `pyproject.toml`:

```toml
[tool.pyright]
disableBytesTypePromotions = true
enableTypeIgnoreComments = true
exclude = [
  ".env",
  ".venv",
  "**/__pycache__",
  "**/.*",
  "**/node_modules",
  "env",
  "local/deprecated",
  "monolith/tests",
  "venv",
]
reportMissingParameterType = true
reportMissingTypeArgument = false
reportUnnecessaryTypeIgnoreComment = "error"
typeCheckingMode = "basic"
```

The error happened in the file `monolith/graphql/warehouse/schema_monitor.py`, which is not in any of the excluded paths. Adding `reportPossiblyUnboundVariable = "error"` makes it so that `pyright` reports the error, as expected.

**Comment by erictraut** [comment]
Pyright is working as intended here. You have configured pyright to use "basic" type checking mode rather than its default of "standard". As you can see from [this table](https://microsoft.github.io/pyright/#/configuration?id=diagnostic-settings-defaults), the `reportPossiblyUnboundVariable` check is disabled by default in "basic" mode.
